### PR TITLE
Deprecate set_manifold(id), moved to reset_manifold(id)

### PR DESCRIPTION
**Pull request by luca-heltai**
#4921

### CODE REVIEW

File: doc/news/changes/minor/20180414LucaHeltai [hunk 1]
```diff
@@ -0,0 +1,8 @@
+New: Introduced Triangulation::reset_manifold(), as a substituted for Triangulation::set_manifold()
```
Comment (jppelteret) [inline doc/news/changes/minor/20180414LucaHeltai]: reset_manifold() as a substituted for

File: doc/news/changes/minor/20180414LucaHeltai [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+with a single argument. The set_manifold() method with one argument only was used to reset the manifold
```
Comment (jppelteret) [inline doc/news/changes/minor/20180414LucaHeltai]: with only one argument

File: doc/news/changes/minor/20180414LucaHeltai [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+object. This method is now deprecated, in favour of the new, more esplicative, method.
```
Comment (jppelteret) [inline doc/news/changes/minor/20180414LucaHeltai]: favor of the new, more explicit method

(note: American spelling of "favour")

File: include/deal.II/grid/tria.h [hunk 4]
```diff
@@ -1769,18 +1768,48 @@ class Triangulation : public Subscriptor
...
+   * Triangulation::reset_manifold() instead.
+   *
    * @see
    * @ref GlossManifoldIndicator "Glossary entry on manifold indicators"
    */
+  DEAL_II_DEPRECATED
   void set_manifold (const types::manifold_id number);
 
+  /**
+   * Reset those parts of the triangulation with the given manifold_id
```
Comment (jppelteret) [inline include/deal.II/grid/tria.h]: @p manifold_id

File: include/deal.II/grid/tria.h [hunk 5]
```diff
@@ -1769,18 +1768,48 @@ class Triangulation : public Subscriptor
...
+   * to use a FlatManifold object. This is the default state of a
+   * non-curved triangulation, and undoes assignment of a different
+   * Manifold object by the function Triangulation::set_manifold()
+   *
+   * @ingroup manifold
+   *
+   * @see
+   * @ref GlossManifoldIndicator "Glossary entry on manifold indicators"
+   */
+  void reset_manifold (const types::manifold_id number);
```
Comment (jppelteret) [inline include/deal.II/grid/tria.h]: number -> manifold_id

(to match the function documentation)

File: include/deal.II/grid/tria.h [hunk 6]
```diff
@@ -1769,18 +1768,48 @@ class Triangulation : public Subscriptor
...
+
+  /**
+   * Reset all parts of the trianuglation, regardless of their
```
Comment (jppelteret) [inline include/deal.II/grid/tria.h]: trianuglation -> triangulation

File: include/deal.II/grid/tria.h [hunk 7]
```diff
@@ -1769,18 +1768,48 @@ class Triangulation : public Subscriptor
...
    * @ref GlossManifoldIndicator "Glossary entry on manifold indicators"
    */
+  DEAL_II_DEPRECATED
   void set_manifold (const types::manifold_id number);
 
+  /**
+   * Reset those parts of the triangulation with the given manifold_id
+   * to use a FlatManifold object. This is the default state of a
+   * non-curved triangulation, and undoes assignment of a different
+   * Manifold object by the function Triangulation::set_manifold()
```
Comment (jppelteret) [inline include/deal.II/grid/tria.h]: Requires a "." at the end of the sentence.

File: doc/news/changes/minor/20180414LucaHeltai (hunk 3 above)
Comment (jppelteret) [reply doc/news/changes/minor/20180414LucaHeltai]: I don't think that you noticed the change for "esplicative". I thought you want to say "explicit", but maybe you meant "explicative".

Comment (luca-heltai) [reply doc/news/changes/minor/20180414LucaHeltai]: I don't know why it's still showing `esplicative`. Latest commit is `explicative`.

Comment (jppelteret) [reply doc/news/changes/minor/20180414LucaHeltai]: I see the updated version now. Sorry about that :-/

**Pull request merged**